pose_estimator/src/pose_estimator.cpp

```diff
@@ -278,22 +285,28 @@ void PoseEstimator::dataCallback(
   orientation_pitch_check_.test(pitch);
   orientation_yaw_check_.test(yaw);
 
-  // Estimate depth from pressure sensor
-  double depth = calculateDepth(pressure_msg->fluid_pressure, in_saltwater_);
-  msg.state.manoeuvring.pose.mean.position.z = depth;
-  msg.state.manoeuvring.pose.covariance[14] = pressure_msg->variance;
+  if (last_pressure_msg_)
+  {
+    // Estimate depth from pressure sensor
+    double depth = calculateDepth(last_pressure_msg_->fluid_pressure, in_saltwater_);
+    msg.state.manoeuvring.pose.mean.position.z = depth;
+    msg.state.manoeuvring.pose.covariance[14] = last_pressure_msg_->variance;
 
-  depth_check_.test(depth);
+    depth_check_.test(depth);
+  }
 
-  // Estimate velocity from AHRS and thruster RPMs
-  const double speed_kn = static_cast<double>(rpms_msg->rpms) / rpm_per_kn_;
-  constexpr double kn_per_ms = 1852. / 3600.;  // exact conversion
-  const tf::Quaternion nominal_velocity(speed_kn * kn_per_ms, 0., 0., 0.);
-  const tf::Quaternion velocity =
+  if (last_rpms_msg_)
+  {
+    // Estimate velocity from AHRS and thruster RPMs
+    const double speed_kn = static_cast<double>(last_rpms_msg_->rpms) / rpm_per_kn_;
+    constexpr double kn_per_ms = 1852. / 3600.;  // exact conversion
```

Comment:
where this number come from?

Reply:
AFAIK 1.852 km/h is the official SI equivalent for nautical knots. I cannot find official documentation, hope [Wikipedia is enough](https://en.wikipedia.org/wiki/Knot_(unit)).

Reply:
SGTM